libraries/AP_RCProtocol/AP_RCProtocol.cpp

```diff
@@ -238,6 +238,27 @@ void AP_RCProtocol::process_handshake( uint32_t baudrate)
   check for bytes from an additional uart. This is used to support RC
   protocols from SERIALn_PROTOCOL
  */
+void AP_RCProtocol::SerialConfig::apply_to_uart(AP_HAL::UARTDriver *uart) const
+{
+    uart->configure_parity(parity);
+    uart->set_stop_bits(stop_bits);
+    uart->set_options(uart->get_options() & ~masked_out_options);
+    uart->set_options(uart->get_options() | masked_in_options);
+    uart->begin(baud, 128, 128);
+}
+
+static const AP_RCProtocol::SerialConfig serial_configs[] {
+    // inverted and uninverted 115200 8N1:
+    { 115200, 0, 1, 0, AP_HAL::UARTDriver::OPTION_RXINV },
```

Comment:
Can you turn parity and stop bits into constants

Reply:
```
    { 115200, AP_RCProtocol::ParityBits::ZERO, AP_RCProtocol::StopBits::ONE, 0, AP_HAL::UARTDriver::OPTION_RXINV },
    { 115200, AP_RCProtocol::ParityBits::ZERO, AP_RCProtocol::StopBits::ONE, AP_HAL::UARTDriver::OPTION_RXINV, 0 },
    // SBUS settings, even parity, 2 stop bits:
    { 100000, AP_RCProtocol::ParityBits::TWO, AP_RCProtocol::StopBits::TWO, AP_HAL::UARTDriver::OPTION_RXINV, 0 },
    // CrossFire:
    { 416666, AP_RCProtocol::ParityBits::ZERO, AP_RCProtocol::StopBits::ONE, 0, AP_HAL::UARTDriver::OPTION_RXINV },
```

yeah, that didn't work out :-)

Do you think a comment will cover it?

```
static const AP_RCProtocol::SerialConfig serial_configs[] {
    // BAUD PRTY STOP MASKED-IN MASKED-OUT
    // inverted and uninverted 115200 8N1:
    { 115200,  0,   1, 0, AP_HAL::UARTDriver::OPTION_RXINV },
    { 115200,  0,   1, AP_HAL::UARTDriver::OPTION_RXINV, 0 },
    // SBUS settings, even parity, 2 stop bits:
    { 100000,  2,   2, AP_HAL::UARTDriver::OPTION_RXINV, 0 },
    // CrossFire:
    { 416666,  0,   1, 0, AP_HAL::UARTDriver::OPTION_RXINV },
};
```

Reply:
Yep fine :)